extensions/sql-bindings/src/common/utils.ts

```diff
@@ -0,0 +1,127 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type * as azdataType from 'azdata';
+import * as vscode from 'vscode';
+import * as vscodeMssql from 'vscode-mssql';
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as glob from 'fast-glob';
+import * as cp from 'child_process';
+
+export interface ValidationResult {
+	errorMessage: string;
+	validated: boolean
+}
+
+/**
+ * Consolidates on the error message string
+ */
+export function getErrorMessage(error: any): string {
+	return (error instanceof Error)
+		? (typeof error.message === 'string' ? error.message : '')
+		: typeof error === 'string' ? error : `${JSON.stringify(error, undefined, '\t')}`;
+}
+
+export async function getAzureFunctionService(): Promise<vscodeMssql.IAzureFunctionsService> {
+	if (getAzdataApi()) {
+		// this isn't supported in ADS
+		throw new Error('Azure Functions service is not supported in Azure Data Studio');
+	} else {
+		const api = await getVscodeMssqlApi();
+		return api.azureFunctions;
+	}
+}
+
+export async function getVscodeMssqlApi(): Promise<vscodeMssql.IExtension> {
+	const ext = vscode.extensions.getExtension(vscodeMssql.extension.name) as vscode.Extension<vscodeMssql.IExtension>;
+	return ext.activate();
+}
+
+export interface IPackageInfo {
+	name: string;
+	fullName: string;
+	version: string;
+	aiKey: string;
+}
+
+export function getPackageInfo(packageJson?: any): IPackageInfo | undefined {
```

Comment:
It'd be better to use vscode.extensions.getExtension(...).packageJson for this instead : https://github.com/microsoft/azuredatastudio/blob/main/src/vs/vscode.d.ts#L6007